tensorrt_llm/_torch/custom_ops/torch_custom_ops.py

```diff
@@ -1035,6 +1042,235 @@ def _(
+        c_tensor = torch.empty(*(m, n), dtype=output_dtype, device="cuda")
+
+        real_k = k * 2
+        sf_m = pad_up(m, 128)
+        sf_k = pad_up(real_k // sf_vec_size, 4)
+        sf_n = pad_up(n, 128)
+
+        a_ptr = make_ptr(
+            # cutlass type
+            cutlass.Float4E2M1FN,
+            a_tensor.data_ptr(),
+            cute.AddressSpace.gmem,
+            assumed_align=32,
+        )
+        b_ptr = make_ptr(
+            cutlass.Float4E2M1FN,
+            b_tensor.data_ptr(),
+            cute.AddressSpace.gmem,
+            assumed_align=32,
+        )
+        a_sf_ptr = make_ptr(
+            cutlass.Float8E4M3FN,
+            a_sf_tensor.data_ptr(),
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        b_sf_ptr = make_ptr(
+            cutlass.Float8E4M3FN,
+            b_sf_tensor.data_ptr(),
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        c_ptr = make_ptr(
+            cutlass.BFloat16,
+            c_tensor.data_ptr(),
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+
+        # get stream
+        torch_stream = torch.cuda.current_stream()
+        stream = cuda.CUstream(torch_stream.cuda_stream)
+
+        gemm_wrapper_func = Sm100BlockScaledPersistentDenseGemmKernelWrapper
+        cache_key = (
+            sf_vec_size,
+            mma_tiler_mn,
+            cluster_shape_mn,
+        )
+        if cache_key not in CuteDSLNVFP4BlackwellLinear.kernel_dict:
+            gemm = gemm_wrapper_func(
+                sf_vec_size,
+                mma_tiler_mn,
+                cluster_shape_mn,
+            )
+            # Compute max active clusters on current device
+            hardware_info = cutlass.utils.HardwareInfo()
+            max_active_clusters = hardware_info.get_max_active_clusters(
+                cluster_shape_mn[0] * cluster_shape_mn[1])
+            hardware_info.get_l2_cache_size_in_bytes()
+
+            compiled_gemm = cute.compile(
+                gemm,
+                m,
+                n,
+                real_k,
+                sf_m // 128,
+                sf_n // 128,
+                sf_k // 4,
+                1,
+                a_ptr,
+                b_ptr,
+                a_sf_ptr,
+                b_sf_ptr,
+                c_ptr,
+                alpha,
+                max_active_clusters,
+                stream,
+            )
+
+            CuteDSLNVFP4BlackwellLinear.kernel_dict[cache_key] = compiled_gemm
+        else:
+            compiled_gemm = CuteDSLNVFP4BlackwellLinear.kernel_dict[cache_key]
+
+        # launch gemm kernel
+        compiled_gemm(m, n, real_k, sf_m // 128, sf_n // 128, sf_k // 4, a_ptr,
+                      b_ptr, a_sf_ptr, b_sf_ptr, c_ptr, alpha, stream)
+        return c_tensor
+
+
+# a/b: fp4, scale: fp8, output: bf16
+@torch.library.custom_op("trtllm::cute_dsl_nvfp4_gemm_blackwell",
```

Comment:
Will this support sm100,sm103,sm120? 

Maybe add come check on the sm version some where.